codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/handler/SockJsSession.java

```diff
@@ -0,0 +1,229 @@
+        return connectionCtx;
+    }
+
+    /**
+     * Sets the {@link ChannelHandlerContext} used to initially connect.
+     *
+     * @param newCtx the ChannelHandlerContext used establishing a connection.
+     */
+    public void setConnectionContext(final ChannelHandlerContext newCtx) {
+        for (;;) {
+            if (CONN_CTX_UPDATER.compareAndSet(this, connectionCtx, newCtx)) {
+                return;
+            }
+        }
+    }
+
+    /**
+     * Returns the {@link ChannelHandlerContext} used on an open session.
+     *
+     * @return {@code ChannelHandlerContext} the ChannelHandlerContext used establishing a connection.
+     */
+    public ChannelHandlerContext openContext() {
+        return openCtx;
+    }
+
+    /**
+     * Sets the {@link ChannelHandlerContext} used to initially connect.
+     *
+     * @param newCtx the ChannelHandlerContext used when the session is open.
+     */
+    public void setOpenContext(final ChannelHandlerContext newCtx) {
+        for (;;) {
+            if (OPEN_CTX_UPDATER.compareAndSet(this, openCtx, newCtx)) {
+                return;
+            }
+        }
+    }
+
+    /**
+     * Sets the {@link State} of this session.
+     *
+     * @param newState the state to which this session should be set.
+     */
+    public void setState(final State newState) {
+        for (;;) {
+            if (STATE_UPDATER.compareAndSet(this, state, newState)) {
+                return;
+            }
+        }
+    }
+
+    public State getState() {
+        return state;
+    }
+
+    public boolean inuse() {
+        return inuse;
+    }
+
+    public void setInuse(final boolean use) {
+        inuse = use;
+    }
+
+    public SockJsConfig config() {
+        return service.config();
+    }
+
+    public String sessionId() {
+        return sessionId;
+    }
+
+    public void onMessage(final String message) throws Exception {
+        service.onMessage(message);
+        updateTimestamp();
+    }
+
+    public void onOpen(final SockJsSessionContext session) {
+        setState(State.OPEN);
+        service.onOpen(session);
+        updateTimestamp();
+    }
+
+    public void onClose() {
+        setState(State.CLOSED);
+        service.onClose();
+    }
+
+    public void addMessage(final String message) {
+        messageQueue.add(message);
+        updateTimestamp();
+    }
+
+    /**
+     * Returns all messages that have been stored in the session.
+     * The messages returned, if any, will be removed from this session.
+     *
+     * @return {@code List} the messages that have been stored in this session.
+     */
+    public List<String> getAllMessages() {
+        final List<String> all = new ArrayList<String>();
+        for (String msg; (msg = messageQueue.poll()) != null;) {
+            all.add(msg);
+        }
+        return all;
+    }
+
+    @SuppressWarnings("ManualArrayToCollectionCopy")
+    public void addMessages(final String[] messages) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)